autoload/test/javascript/mocha.vim

```diff
@@ -4,6 +4,7 @@ endif
 
 function! test#javascript#mocha#test_file(file) abort
   return a:file =~# g:test#javascript#mocha#file_pattern
+	  \ && (test#javascript#has_package('mocha') || !empty(test#javascript#mocha#executable()))
```

Comment:
In some JavaScript test runners like `mocha`, `test#javascript#{runner}#executable()` will fallback to a global executable if it doesn't find the one in `node_modules/.bin`, so the `!empty(test#javascript#mocha#executable())` conditional will always evaluate to true.

However, for the sake of consistency we can leave it as it is, and remove it once we implement https://github.com/janko-m/vim-test/issues/185. To clarify, after https://github.com/janko-m/vim-test/issues/185 I plan that we stop activating a JavaScript runner if only the pattern matches and the global executable exists (without it existing in `package.json`). Considering tha then users will be able to selecting runners manually (instead of having to rely on test.vim to always correctly detect it), we can require that pattern matches and the runner is present in `package.json`, or the runner is hardcoded via a variable.

Just wanted to have a stream of thought, nothing to change here now.